tests/Aspire.Hosting.Containers.Tests/ContainerResourceTests.cs

```diff
@@ -185,7 +185,7 @@ public async Task EnsureContainerWithVolumesEmitsVolumes()
         builder.AddContainer("containerwithvolumes", "image/name")
             .WithVolume("myvolume", "/mount/here")
             .WithVolume("myreadonlyvolume", "/mount/there", isReadOnly: true)
-            .WithVolume(null! /* anonymous volume */, "/mount/everywhere");
+            .WithVolume("/mount/everywhere");
```

Comment:
Should we update the API so allow for `name` to be null? If you follow the call path, it gets passed into `source` below:

```
    /// <param name="source">The source path if a bind mount or name if a volume. Can be <c>null</c> if the mount is an anonymous volume.</param>
    /// <param name="target">The target path of the mount.</param>
    /// <param name="type">The type of the mount.</param>
    /// <param name="isReadOnly">A value indicating whether the mount is read-only.</param>
    public ContainerMountAnnotation(string? source, string target, ContainerMountType type, bool isReadOnly)
```

Also, these `WithVolume` APIs don't meet .NET Design Guidelines:

https://github.com/dotnet/aspire/blob/4e7e0de000bb83f824cca2773892a22b5da89245/src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs#L50

https://github.com/dotnet/aspire/blob/4e7e0de000bb83f824cca2773892a22b5da89245/src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs#L63

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/member-overloading

❌ AVOID being inconsistent in the ordering of parameters in overloaded members. Parameters with the same name should appear in the same position in all overloads.

Reply:
Should we update the API so allow for name to be null?

Reply:
Because it makes it easier to write code like this:

```csharp
string? volumeName = GetVolumeName(); // may return null

builder.AddContainer("containerwithvolumes", "image/name")
    .WithVolume(volumeName, "/mount/everywhere");
```

Reply:
See also: https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/member-overloading

✔️ DO allow null to be passed for optional arguments.

Reply:
Ok